tests/foreman/ui/test_discoveryrule.py

```diff
@@ -172,11 +351,266 @@ def test_positive_update_name(self):
+            self.assertIsNotNone(self.discoveryrules.search(name))
+            self.assertEqual(
+                self.discoveryrules.get_attribute_value(
+                    name, 'hostgroup', element_type='select'),
+                self.host_group.name
+            )
+            self.discoveryrules.update(name=name, hostgroup=new_hostgroup_name)
+            self.assertEqual(
+                self.discoveryrules.get_attribute_value(
+                    name, 'hostgroup', element_type='select'),
+                new_hostgroup_name
+            )
+
+    @run_only_on('sat')
+    @tier1
+    def test_positive_update_hostname(self):
+        """Update discovery rule hostname value
+
+        @Feature: Discovery Rule - Update
+
+        @Assert: Rule host name is updated
+        """
+        name = gen_string('alpha')
+        hostname = gen_string('alpha')
+        with Session(self.browser) as session:
+            make_discoveryrule(
+                session, name=name, hostgroup=self.host_group.name)
+            self.assertIsNotNone(self.discoveryrules.search(name))
+            self.discoveryrules.update(name=name, hostname=hostname)
+            self.assertEqual(
+                self.discoveryrules.get_attribute_value(name, 'hostname'),
+                hostname
+            )
+
+    @run_only_on('sat')
+    @tier1
+    def test_positive_update_limit(self):
+        """Update discovery rule limit value
+
+        @Feature: Discovery Rule - Update
+
+        @Assert: Rule host limit field is updated
+        """
+        name = gen_string('alpha')
+        limit = str(gen_integer(1, 100))
+        with Session(self.browser) as session:
+            make_discoveryrule(
+                session, name=name, hostgroup=self.host_group.name)
+            self.assertIsNotNone(self.discoveryrules.search(name))
+            self.discoveryrules.update(name=name, host_limit=limit)
+            self.assertEqual(
+                self.discoveryrules.get_attribute_value(name, 'host_limit'),
+                limit
+            )
+
+    @run_only_on('sat')
+    @tier1
+    def test_positive_update_priority(self):
+        """Update discovery rule priority value
+
+        @Feature: Discovery Rule - Update
+
+        @Assert: Rule priority is updated
+        """
+        name = gen_string('alpha')
+        priority = str(gen_integer(1, 100))
+        with Session(self.browser) as session:
+            make_discoveryrule(
+                session, name=name, hostgroup=self.host_group.name)
+            self.assertIsNotNone(self.discoveryrules.search(name))
+            self.discoveryrules.update(name=name, priority=priority)
+            self.assertEqual(
+                self.discoveryrules.get_attribute_value(name, 'priority'),
+                priority
+            )
+
+    @run_only_on('sat')
+    @tier1
+    def test_positive_update_disable_enable(self):
+        """Update discovery rule enabled state. (Enabled->Disabled->Enabled)
```

Comment:
Probably, it will be more logical. I just thought to emphasize the fact that I am changing from default Enabled to Disabled state for create procedure